packages/contracts/hardhat.config.ts

```diff
@@ -108,6 +110,30 @@ const config: HardhatUserConfig = {
   etherscan: {
     apiKey: process.env.ETHERSCAN_API_KEY,
   },
+  dodoc: {
+    runOnCompile: true,
+    exclude: [
+      'Helper_GasMeasurer',
+      'Helper_SimpleProxy',
+      'TestERC20',
+      'TestLib_CrossDomainUtils',
+      'TestLib_OVMCodec',
+      'TestLib_RLPReader',
+      'TestLib_RLPWriter',
+      'TestLib_AddressAliasHelper',
+      'TestLib_MerkleTrie',
+      'TestLib_SecureMerkleTrie',
+      'TestLib_Buffer',
+      'TestLib_Bytes32Utils',
+      'TestLib_BytesUtils',
+      'TestLib_MerkleTree',
+    ],
+  },
+  outputValidator: {
+    runOnCompile: true,
+    errorMode: true,
```

Comment:
Happy to hear ! I'll continue extending with anything you need. 

On your points: 
1. @inheritdoc should be working fine. I belive that the problem is with the state variables in `CrossDomainEnabled`.
2. Events = done. Public variables however are essentially functions in the abi & I can't differentiate. I'll extend the package to check in the "ast tree" for this case.
3. Should be fine now. 

* I've **disabled the check for user docs** since you're not using it anywhere (@notice). Should I return it ? 
* I've made the plugin "less-strict" accepting any Natspect comments. (config option - strict: false)

Next: 
1. @indeavr disable public vars. 
2. @elenadimitrova (after 1) Check for anything else & also review the contracts/docs folder for the actual MD output :)

Reply:
Pushed ! 
The "variables: false" config option is now available. 

However be aware that there is still one issue -> "ignore state variables in base contracts". Currently only in `Lib_AddressResolver`.

I'll be doing more testing ! 
Let me know if you find any other issues :)

Reply:
Very nice thanks!
How difficult is it to only show a single instance of a missing natspec in base contract which is inherited elsewhere and therefore raises the error multiple times, even though only one natspec is missing. e.g. `xDomainMessageSender` is not documented in `ICrossDomainMessenger` but errors are logged from `IL1CrossDomainMessenger.sol`, `IL2CrossDomainMessenger.sol`, `L1CrossDomainMessenger.sol` and `L2CrossDomainMessenger.sol` which all inherit it.

Reply:
> Very nice thanks! How difficult is it to only show a single instance of a missing natspec in base contract which is inherited elsewhere and therefore raises the error multiple times, even though only one natspec is missing. e.g. `xDomainMessageSender` is not documented in `ICrossDomainMessenger` but errors are logged from `IL1CrossDomainMessenger.sol`, `IL2CrossDomainMessenger.sol`, `L1CrossDomainMessenger.sol` and `L2CrossDomainMessenger.sol` which all inherit it.

I'll need to add some form of memoization.

Reply:
Don't worry if it's not straightforward. You can add it on the wish list in `hardhat-output-validator`

Reply:
👍 Recursive ("inherited contracts") support for variable exclusion is now supported by the plugin
  
👎 Memoization for the output can be added on a later stage since I feel it is a lot of effort for little benefit. (prove me wrong & I'll add it now :) )

Reply:
What is the difference between Recursive and Memoization?

Reply:
Sorry my comments were unclear ! 

You can interpret "Recursive" as `variables exclusion now works properly :D`
And the "Memoization" was meant to answer:
> How difficult is it to only show a single instance of a missing natspec in base contract ?

Answer: Quite hard.